src/test/scala/com/microsoft/hyperspace/index/LogicalPlanSerDeTests.scala

```diff
@@ -69,10 +73,52 @@ class LogicalPlanSerDeTests extends SparkFunSuite with SparkInvolvedSuite {
     StructType(attributes.map(a => StructField(a.name, a.dataType, a.nullable, a.metadata)))
   }
 
-  test("Serde query with Hadoop file system relation.") {
+  test("Serde query with Hadoop file system parquet relation.") {
     verifyPlanSerde(scanNode, "hadoopFsRelation.plan")
   }
 
+  test("Serde query with Hadoop file system csv relation.") {
+    val csvFormat = new CSVFileFormat
+    val relation: HadoopFsRelation =
+      scanNode.relation.asInstanceOf[HadoopFsRelation].copy(fileFormat = csvFormat)(spark)
+    val csvScanNode = scanNode.copy(relation = relation)
+
+    // Csv file format is serializable unless isSplittable is called on it. isSplittable api
+    // initializes internal objects which break serialization logic.
+    val kryoSerializer = new KryoSerializer(spark.sparkContext.getConf)
+    KryoSerDeUtils.serialize(kryoSerializer, csvFormat)
+
+    // Confirm that isSplittable makes serialization fail
+    intercept[KryoException] {
+      csvFormat.isSplitable(spark, Map(), new Path("path"))
+      KryoSerDeUtils.serialize(kryoSerializer, csvFormat)
+    }
+
+    // Now verify if Hyperspace serialization still works with csv format
+    verifyPlanSerde(csvScanNode, "hadoopFsRelation.plan")
+  }
+
+  test("Serde query with Hadoop file system json relation.") {
+    val jsonFormat = new JsonFileFormat
+    val relation: HadoopFsRelation =
+      scanNode.relation.asInstanceOf[HadoopFsRelation].copy(fileFormat = jsonFormat)(spark)
+    val csvScanNode = scanNode.copy(relation = relation)
```

Comment:
looks like we can refactor most of these code? (btw, it should be json here)

Reply:
thanks @imback82 , do you mean deduplication?
is this how you mean?
```
def verifySerdeFailsForFormat(format): Unit = {
  kyro.serialize()
  intercept[Exception] {
     format.isSplittable
     kyro.serialize()
   }
}
```
and then use it for csv and json?

Reply:
thanks @imback82 , do you mean deduplication?
is this how you mean?
```
def verifySerdeFailsForFormat(format): Unit = {
  kyro.serialize()
  intercept[Exception] {
     format.isSplittable
     kyro.serialize()
   }
}
```
and then use it for csv and json?

Reply:
updated